urllib3/response.py

```diff
@@ -56,14 +56,30 @@ class GzipDecoder(object):
 
     def __init__(self):
         self._obj = zlib.decompressobj(16 + zlib.MAX_WBITS)
+        self._state = 'first'
```

Comment:
If we're going with an arbitrary set of states to move between, let's create an Enum higher up to list those states comprehensively, and then use the appropriate namespaced variables, rather than free-form strings.

Reply:
I actually consider using string for these simple state changes more readable.  In languages like Java where we care about the last drop of performance we don't like such strings.  But in Python I tends to use plain strings for such three-state variables.

But since you like it the other way round, and since I don't really care, it is in the new commit.